### PR TITLE
disttask: rename dispatcher to scheduler

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/49125

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/backfilling_dispatcher_test.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func TestBackfillingDispatcherGlobalSortMode(t *testing.T) {
 	ctx = util.WithInternalSourceType(ctx, "handle")
 	mgr := storage.NewTaskManager(pool)
 	storage.SetTaskManager(mgr)
-	dspManager, err := dispatcher.NewManager(util.WithInternalSourceType(ctx, "dispatcher"), mgr, "host:port")
+	dspManager, err := scheduler.NewManager(util.WithInternalSourceType(ctx, "scheduler"), mgr, "host:port")
```
Comment (ywqzzy) [inline pkg/ddl/backfilling_dispatcher_test.go:156]:
```suggestion
	scheManager, err := scheduler.NewManager(util.WithInternalSourceType(ctx, "scheduler"), mgr, "host:port")
```

File: pkg/ddl/backfilling_dispatcher.go [hunk 2]
```diff
@@ -42,33 +42,33 @@ import (
...
 
-// OnTick implements dispatcher.Extension interface.
-func (*BackfillingDispatcherExt) OnTick(_ context.Context, _ *proto.Task) {
+// OnTick implements scheduler.Extension interface.
+func (*BackfillingSchedulerExt) OnTick(_ context.Context, _ *proto.Task) {
 }
 
 // OnNextSubtasksBatch generate batch of next step's plan.
-func (dsp *BackfillingDispatcherExt) OnNextSubtasksBatch(
+func (dsp *BackfillingSchedulerExt) OnNextSubtasksBatch(
```
Comment (ywqzzy) [inline pkg/ddl/backfilling_dispatcher.go:69]:
```suggestion
func (se *BackfillingSchedulerExt) OnNextSubtasksBatch(
```
change the receiver name

File: pkg/disttask/framework/scheduler/rebalance_test.go [hunk 3]
```diff
@@ -60,10 +60,10 @@ func scaleTest(t *testing.T,
 	if len(testCase.cleanedNodes) > 0 {
 		mockTaskMgr.EXPECT().GetSubtasksByExecIdsAndStepAndState(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil)
 	}
-	dsp := dispatcher.NewBaseDispatcher(ctx, mockTaskMgr, "server", &proto.Task{Step: proto.StepInit, ID: int64(id)})
+	dsp := scheduler.NewBaseScheduler(ctx, mockTaskMgr, "server", &proto.Task{Step: proto.StepInit, ID: int64(id)})
```
Comment (ywqzzy) [inline pkg/disttask/framework/scheduler/rebalance_test.go:63]: ditto

File: pkg/disttask/framework/scheduler/scheduler_manager.go [hunk 4]
```diff
@@ -51,11 +51,11 @@ func (dm *Manager) getRunningTaskCnt() int {
 	return len(dm.runningTasks.taskIDs)
 }
 
-func (dm *Manager) setRunningTask(task *proto.Task, dispatcher Dispatcher) {
+func (dm *Manager) setRunningTask(task *proto.Task, scheduler Scheduler) {
 	dm.runningTasks.Lock()
```
Comment (ywqzzy) [inline pkg/disttask/framework/scheduler/scheduler_manager.go:55]: sm

File: pkg/disttask/framework/scheduler/scheduler.go [hunk 5]
```diff
@@ -125,24 +125,24 @@ func NewBaseDispatcher(ctx context.Context, taskMgr TaskManager, serverID string
...
 
-// Close closes the dispatcher.
-func (*BaseDispatcher) Close() {
+// Close closes the scheduler.
+func (*BaseScheduler) Close() {
 }
 
 // refreshTask fetch task state from tidb_global_task table.
-func (d *BaseDispatcher) refreshTask() error {
+func (d *BaseScheduler) refreshTask() error {
```
Comment (ywqzzy) [inline pkg/disttask/framework/scheduler/scheduler.go:145]:
```suggestion
func (s *BaseScheduler) refreshTask() error {
```

File: pkg/disttask/framework/storage/task_table.go [hunk 6]
```diff
@@ -975,7 +975,7 @@ func (stm *TaskManager) TransferTasks2History(ctx context.Context, tasks []*prot
 	return stm.WithNewTxn(ctx, func(se sessionctx.Context) error {
 		insertSQL := new(strings.Builder)
 		if err := sqlescape.FormatSQL(insertSQL, "replace into mysql.tidb_global_task_history"+
-			"(id, task_key, type, dispatcher_id, state, priority, start_time, state_update_time,"+
+			"(id, task_key, type, scheduler_id, state, priority, start_time, state_update_time,"+
```
Comment (ywqzzy) [inline pkg/disttask/framework/storage/task_table.go:978]: won't pass the test

File: pkg/disttask/importinto/scheduler_test.go [hunk 7]
```diff
@@ -61,7 +61,7 @@ func (s *importIntoSuite) TestDispatcherGetEligibleInstances() {
 	}
 	mockedAllServerInfos := makeFailpointRes(serverInfoMap)
 
-	dsp := ImportDispatcherExt{}
+	dsp := ImportSchedulerExt{}
```
Comment (ywqzzy) [inline pkg/disttask/importinto/scheduler_test.go:64]: ditto

File: pkg/session/bootstrap.go [hunk 8]
```diff
@@ -573,7 +573,7 @@ const (
 		id BIGINT(20) NOT NULL AUTO_INCREMENT PRIMARY KEY,
     	task_key VARCHAR(256) NOT NULL,
 		type VARCHAR(256) NOT NULL,
-		dispatcher_id VARCHAR(256),
+		scheduler_id VARCHAR(256),
```
Comment (okJiang) [inline pkg/session/bootstrap.go:576]: the table field will update later, better not in this pr

**Review by ywqzzy — APPROVED** [review]

File: pkg/errno/logredaction.md [hunk 9]
```diff
@@ -25,9 +25,9 @@ ERROR 1062 (23000): Duplicate entry '?' for key '?'
 
 And its corresponding log is：
 ```
-[2020/10/20 11:45:37.796 +08:00] [INFO] [conn.go:800] ["command dispatched failed"] [conn=5] [connInfo="id:5, addr:127.0.0.1:57222 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="insert into t values (1),(1)"] [txn_mode=OPTIMISTIC] [err="[kv:1062]Duplicate entry '1' for key 'a'"]
+[2020/10/20 11:45:37.796 +08:00] [INFO] [conn.go:800] ["command scheduled failed"] [conn=5] [connInfo="id:5, addr:127.0.0.1:57222 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="insert into t values (1),(1)"] [txn_mode=OPTIMISTIC] [err="[kv:1062]Duplicate entry '1' for key 'a'"]
```
Comment (D3Hunter) [inline pkg/errno/logredaction.md:28]: this is not part of disttask

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: pkg/ddl/job_table.go [hunk 10]
```diff
@@ -50,13 +50,13 @@ import (
 
 var (
 	addingDDLJobConcurrent      = "/tidb/ddl/add_ddl_job_general"
-	dispatchLoopWaitingDuration = 1 * time.Second
```
Comment (D3Hunter) [inline pkg/ddl/job_table.go:53]: seems this is not part of framework

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**